File: app/src/main/java/com/maiya/leetcode/util/file/download/FileDownloadRequest.java
package com.maiya.leetcode.util.file.download;

import android.os.AsyncTask;
import android.text.TextUtils;

import java.io.File;


public class FileDownloadRequest {

    public static void download(String url, File target, int type, FileDownloadCallback callback) {
        if (!TextUtils.isEmpty(url) && target != null) {
            FileDownloadTask task = new FileDownloadTask(url, target, type, callback);
            task.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
        }
    }
}

File: app/src/main/java/com/maiya/call/util/file/download/FileDownloadRequest.kt
package com.maiya.call.util.file.download

import android.os.AsyncTask
import android.text.TextUtils
import java.io.File

object FileDownloadRequest {
    fun download(url: String?, target: File?, type: Int, callback: FileDownloadCallback?) {
        if (!TextUtils.isEmpty(url) && target != null) {
            val task = FileDownloadTask(url, target, type, callback)
            task.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR)
        }
    }
}